codegen/src/main/java/software/amazon/awssdk/codegen/poet/model/PaginatorResponseClassSpec.java

```diff
@@ -0,0 +1,528 @@
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.codegen.poet.model;
+
+import com.squareup.javapoet.ClassName;
+import com.squareup.javapoet.CodeBlock;
+import com.squareup.javapoet.FieldSpec;
+import com.squareup.javapoet.MethodSpec;
+import com.squareup.javapoet.ParameterizedTypeName;
+import com.squareup.javapoet.TypeName;
+import com.squareup.javapoet.TypeSpec;
+import java.security.InvalidParameterException;
+import java.util.Iterator;
+import java.util.List;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import javax.lang.model.element.Modifier;
+import software.amazon.awssdk.codegen.model.intermediate.IntermediateModel;
+import software.amazon.awssdk.codegen.model.intermediate.MemberModel;
+import software.amazon.awssdk.codegen.model.intermediate.OperationModel;
+import software.amazon.awssdk.codegen.model.intermediate.ShapeModel;
+import software.amazon.awssdk.codegen.model.service.PaginatorDefinition;
+import software.amazon.awssdk.codegen.poet.ClassSpec;
+import software.amazon.awssdk.codegen.poet.PoetExtensions;
+import software.amazon.awssdk.codegen.poet.PoetUtils;
+import software.amazon.awssdk.pagination.NextPageSupplier;
+import software.amazon.awssdk.pagination.Paginated;
+import software.amazon.awssdk.pagination.PaginatedItemsIterable;
+import software.amazon.awssdk.pagination.PaginatedResponsesIterator;
+import software.amazon.awssdk.pagination.SdkIterable;
+
+/**
+ * Java poet {@link ClassSpec} to generate the response class for sync paginated operations.
+ *
+ * Sample of a generated class with annotations:
+
+    public final class ListTablesPaginator implements Paginated&lt;ListTablesResponse, String> {
+        private final DynamoDBClient client;
+
+        private final ListTablesRequest firstRequest;
+
+        private final ListTablesResponse firstResponsePage;
+
+        private final NextPageSupplier nextPageSupplier;
+
+        public ListTablesPaginator(final DynamoDBClient client, final ListTablesRequest firstRequest,
+            final ListTablesResponse firstResponsePage) {
+            this.client = client;
+            this.firstRequest = firstRequest;
+            this.firstResponsePage = firstResponsePage;
+            this.nextPageSupplier = new ListTablesResponseSupplier();
+        }
+
+        public ListTablesResponse firstPage() {
+            return firstResponsePage;
+        }
+
+        public Iterator&lt;ListTablesResponse> iterator() {
+            return new PaginatedResponsesIterator(firstResponsePage, nextPageSupplier);
+        }
+
+        public SdkIterable&lt;String> tableNames() {
+            Function&lt;ListTablesResponse, Iterator&lt;String>> getPaginatedMemberIterator = response -> response != null
+                            ? response.tableNames().iterator() : null;
```

Comment:
null?

Reply:
If response is null, I am returning null. This is used in PaginatedItemsIterable which checks has a check to handle null values. Are you suggesting to use empty iterator?